src/components/Chat/MessageAttachment.tsx

```diff
@@ -48,45 +53,53 @@ export const MessageAttachment = ({ attachment, isInner }: Props) => {
               />
             </a>
           ) : (
-            <IconPaperclip
-              size={18}
-              className="shrink-0 text-gray-500 hover:text-blue-500"
-            />
+            <IconPaperclip size={18} className="shrink-0 text-gray-500" />
           )}
         </div>
         <button
           onClick={() => {
-            setIsOpened((isOpened) => !isOpened);
+            setIsExpanded((isExpanded) => !isExpanded);
+            if (!attachment.url) {
+              setIsOpened((isOpened) => !isOpened);
+            }
           }}
           className="flex grow items-center justify-between overflow-hidden"
         >
           <span
             className={`shrink text-left text-sm ${
-              isOpened ? 'max-w-full' : 'max-w-[calc(100%-30px)] truncate'
+              isExpanded ? 'max-w-full' : 'max-w-[calc(100%-30px)] truncate'
             }`}
             title={attachment.title}
           >
             {attachment.title || t('Attachment')}
           </span>
-          <ChevronDown
-            height={18}
-            width={18}
-            className={`shrink-0 text-gray-500 transition ${
-              isOpened ? 'rotate-180' : ''
-            }`}
-          />
+          {isOpenable ? (
+            <ChevronDown
+              height={18}
+              width={18}
+              className={`shrink-0 text-gray-500 transition ${
+                isOpened ? 'rotate-180' : ''
+              }`}
+            />
+          ) : (
+            <a
+              download={attachment.title}
+              href={`api/files?path=${attachment.url}`}
+              onClick={stopBubbling}
+              className="text-gray-500 hover:text-blue-500"
+            >
+              <IconDownload size={18} />
+            </a>
+          )}
         </button>
       </div>
-      {(attachment.data || attachment.url) && isOpened && (
+      {isOpenable && attachment.data && isOpened && (
         <div
           className={`relative mt-2 h-auto w-full overflow-hidden p-3 pt-4 text-sm duration-200`}
         >
           {imageTypes.includes(attachment.type) ? (
```

Comment:
it would be nice to save `imageTypes` in constant outside the component(new array each render) and convert it to `Set` (O(n)->O(1))
```
const imageTypes: ImageMIMEType[] = new Set(['image/jpeg', 'image/png']);
```
and use
```suggestion
          {imageTypes.has(attachment.type) ? (
```